add CheckDivide

ShACKAL762 opened the pull request:
Первый проект на Яндекс Пакатикум.

ShACKAL762 committed:
add CheckDivide

vlad-zotov commented on src/main/java/CheckDivide.java:
```diff
@@ -0,0 +1,83 @@
+public class CheckDivide {
+    public CheckDivide(){
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
+        order(bufferedReader, members(bufferedReader));
```
Привет. Ты можешь передавать в `order()` только один аргумент `bufferedReader`, а `members(bufferedReader)` делать внутри `order()`

vlad-zotov commented on src/main/java/CheckDivide.java:
```diff
@@ -0,0 +1,83 @@
+            try {
+
+                inputLine = bufferedReader.readLine();
+                if (inputLine.trim().equalsIgnoreCase("завершить")){
```
`"завершить"` будет удобнее вынести в константы. Это позволит расширять число стоп слов менее болезненно.

vlad-zotov commented on src/main/java/CheckDivide.java:
```diff
@@ -0,0 +1,83 @@
+                    if (Double.parseDouble(buffer[1]) <0.01)
+                        throw new NumberFormatException();
+                    sum = sum + Double.parseDouble(buffer[1]);
+                    check.append("\n").append(buffer[0]).append(" ").append(buffer[1]);
```
для улучшения читаемости кода советую размещать каждый `append` на отдельной строке

vlad-zotov commented on src/main/java/RubFormat.java:
```diff
@@ -0,0 +1,12 @@
+        int x = (int) sum % 10;
+        return switch (x) {
+            case 1 -> "рубль.";
+            case 2, 3, 4 -> " рубля.";
```
определись где ты ставишь пробел перед словом `рубль` здесь в форматтере или выше в коде. Я рекомендую ставить пробел выше в коде, где форматируется итоговая строка, а здесь возвращать слово `рубль` без пробела

ShACKAL762 committed:
Перенес вызвов members(BufferedReader) в тело метода order(BufferedReader);
Перенёс "завершить" в final String CHECK_CLOSE;
Поправил пробелы в "Рублях";
Сделал перенос строк check.append();//Это был авторефакторинг

ShACKAL762 committed:
Исправил форматирование для 11-14 рублей
